src/main/java/com/googlecode/objectify/impl/TypeUtils.java

```diff
@@ -30,13 +35,20 @@
 	private TypeUtils() {
 	}
 
+	private static final Map<Class<?>, Constructor<?>> noArgConstructorCache = Maps.newConcurrentMap();
+	
 	/**
 	 * Throw an IllegalStateException if the class does not have a no-arg constructor.
 	 */
 	public static <T> Constructor<T> getNoArgConstructor(Class<T> clazz) {
 		try {
-			Constructor<T> ctor = clazz.getDeclaredConstructor(new Class[0]);
-			ctor.setAccessible(true);
+			@SuppressWarnings("unchecked")
+			Constructor<T> ctor = (Constructor<T>) noArgConstructorCache.get(clazz);
```

Comment:
The map is thread safe, but in theory you can compute the cache value more than once. `Map.computeIfAbsent()` would solve this.

Reply:
Java 7. No such method here.